spec/sanity/deployments.js

```diff
@@ -14,42 +14,80 @@ describe('deployments page', function () {
     });
 
 
+    describe('deleting deployment', function(){
+
+        var deploymentName;
+        var deploymentsOpts;
+
+        beforeEach(function(){
+            deploymentName ='deployment_' + new Date().getTime();
+            deploymentsOpts = {id : deploymentName };
+            // lets deploy a blueprint to make this test rerunable.. I hate it, but what can we do
+            // if deploying was a simple act, we would not care..
+            components.ui.layout.goToBlueprints();
+            components.ui.blueprints.IndexPage.createDeployment({ name : 'nodecellar1' });
+            components.ui.blueprints.CreateDeployment.setDetails({ name : deploymentName , 'raw' : {
+                'agent_private_key_path': 'ggg',
+                'agent_user': 'ggg',
+                'host_ip': 'ggg'
+            }});
+            components.ui.blueprints.CreateDeployment.confirm();
+
+            components.ui.deployments.DeploymentPage.waitForInitializingToStop();
+            browser.sleep(1000).then();
 
-    it('should delete a deployment', function (done) {
-        logger.trace('start create deployment test');
+        });
 
-        var deploymentName ='deployment_' + new Date().getTime();
-        // lets deploy a blueprint to make this test rerunable.. I hate it, but what can we do
-        // if deploying was a simple act, we would not care..
-        components.ui.layout.goToBlueprints();
-        components.ui.blueprints.IndexPage.createDeployment({ name : 'nodecellar1' });
-        components.ui.blueprints.CreateDeployment.setDetails({ name : deploymentName , 'raw' : {
-            'agent_private_key_path': 'ggg',
-            'agent_user': 'ggg',
-            'host_ip': 'ggg'
-        }});
-        components.ui.blueprints.CreateDeployment.confirm();
+        it('should work from deployments table view', function (done) {
+            logger.trace('start create deployment test');
+
+            components.ui.layout.goToDeployments();
+            browser.sleep(1000).then();
+
+            var deploymentsOpts = {id : deploymentName };
+            components.ui.deployments.IndexPage.deleteDeployment(deploymentsOpts);
+            components.ui.deployments.DeleteDeployment.clickCancel();
+            browser.sleep(1000);
```

Comment:
so I would put it in "DeleteDeployment.clickCancel()" since everyone who uses that function will have to call sleep right after, and it's implementation specific - if sleep was without a dialog, it would not need sleep.